bin/src/command/orders.rs

```diff
@@ -350,6 +353,70 @@ impl CommandServer {
         Ok(())
     }
 
+    pub async fn list_frontends(
+        &mut self,
+        client_id: String,
+        message_id: &str,
+        filters: FrontendFilters,
+    ) -> anyhow::Result<()> {
+        info!(
+            "Received a request to list frontends, along these filters: {:#?}",
+            filters
+        );
+
+        info!("Here are the filters: {:#?}", filters);
+
+        // if no http / https / tcp filter is provided, list all of them
+        let list_all = !filters.http && !filters.https && !filters.tcp;
+
+        let filter_by_domain = filters.domain.is_some();
+
+        let mut listed_frontends = ListedFrontends::default();
+
+        if filters.http || list_all {
+            for http_frontend in self.state.http_fronts.iter().filter(|f| {
+                if filter_by_domain {
```

Comment:
You should use a destructuring if. It will look like the following example:

```rust
if let Some(domain) = &filters.domain {
    f.1.hostname.contains(domain)
} else {
    false
}
```